electrum/gui/qt/wizard/wizard.py

```diff
@@ -0,0 +1,233 @@
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+from PyQt5.QtCore import Qt, QTimer, pyqtSignal, pyqtSlot, QSize
+from PyQt5.QtGui import QPixmap
+from PyQt5.QtWidgets import (QDialog, QApplication, QPushButton, QWidget, QLabel, QVBoxLayout, QScrollArea,
+                             QHBoxLayout, QLayout, QStackedWidget)
+
+from electrum.i18n import _
+from ..util import Buttons, icon_path
+from electrum.logging import get_logger
+
+if TYPE_CHECKING:
+    from electrum.simple_config import SimpleConfig
+
+
+class QEAbstractWizard(QDialog):
+    _logger = get_logger(__name__)
```

Comment:
I think it is nicer to inherit Logger, e.g. `QEAbstractWizard(QDialog, Logger)`, and then
```py
def __init__(self, config: 'SimpleConfig', app: QApplication, daemon):
    Logger.__init__(self)
```

Especially as `QEAbstractWizard` is not a singleton, it could be useful for debugging to have separate instances of it each their own logger. E.g. look at wallet.py/Abstract_Wallet: note it is overriding Logger.diagnostic_name. -- Though if you don't override diagnostic_name, the choice of having a per-class/per-instance logger becomes less important.

I guess this remark also applies to a lot of existing code in the qml GUI.
Anyway, it's more of a nit.

Reply:
Thanks. I guess this stems from my past experience with C++ and Java, where inheritance is more rigid. Pythonic multiple inheritance and mixins still feel a bit 'loose' to me :)

Reply:
BTW, `diagnostic_name` might not be unique unless I use the `id()` of the wizard instance.

Reply:
> I'm used to loggers being a composition into a class


You are right that inheritance should not be overused.
Hmm, I guess it is mostly more consistent usage of the Logger across the codebase that I wanted, not necessarily inheritance.
We could change the Logger class to be used with composition instead. `diagnostic_name` and `LOGGING_SHORTCUT` could be passed as parameters to `Logging.__init__`,
and then do e.g. `self.logger = Logger(name=self.diagnostic_name(), shortcut=LOGGING_SHORTCUT)` in init.


> `diagnostic_name` might not be unique unless I use the `id()` of the wizard instance

Yes, indeed. See `lnpeer.Peer` for an example where we explicitly use `id()` (through the transport) -- it was added during/after debugging an issue where multiple Peer objects did have the same colliding diagnostic names which was causing confusion.

---

Anyway, it's just a small remark; not too important re this PR.